job.py

```diff
@@ -0,0 +1,85 @@
+import shutil
+from urllib.request import urlopen
+from pathlib import Path
+import ssl
+import json
+from logger import logger
+
+
+def coroutine(f):
+    def wrap(*args, **kwargs):
+        gen = f(*args, **kwargs)
+        gen.send(None)
+        return gen
+    return wrap
+
+
+def get_and_write_data(condition, url):
+    context = ssl._create_unverified_context()
+    file = 'punchline.txt'
+    with condition:
+        with urlopen(url, context=context) as req:
```

Comment:
как насчет асинхронности?) Используем [aiohttp](https://docs.aiohttp.org/en/stable/client_quickstart.html):
```suggestion
        import aiohttp
        async with aiohttp.ClientSession() as session:
            async with session.get(url) as resp:
                print(resp.status)
                print(await resp.text())
```